app/pages/reset-password.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { PropTypes } from 'react';
+import auth from 'panoptes-client/lib/auth';
+import Translate from 'react-translate-component';
+import alert from '../lib/alert';
+import LoginDialog from '../partials/login-dialog';
+
+class ResetPasswordPage extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      inProgress: false,
+      resetSuccess: false,
+      resetError: null,
+      emailSuccess: false,
+      emailError: false,
+      emailIsValid: false
+    };
+
+    this.handlePasswordResetSubmit = this.handlePasswordResetSubmit.bind(this);
+    this.handleEmailSubmit = this.handleEmailSubmit.bind(this);
+    this.handleEmailChange = this.handleEmailChange.bind(this);
+  }
+
+  handleEmailChange(event) {
+    this.setState({ emailIsValid: event.target.checkValidity() });
+  }
+
+  handleEmailSubmit(event) {
+    event.preventDefault();
+
+    this.setState({
+      inProgress: true,
+      emailSuccess: false,
+      emailError: null
+    });
+
+    const email = this.email.value;
+
+    auth.requestPasswordReset({ email })
+      .then(() => {
+        this.setState({
+          emailSuccess: true,
+          inProgress: false
+        });
+      })
+      .catch((error) => {
+        this.setState({ emailError: error });
+      });
+  }
+
+  handlePasswordResetSubmit(event) {
+    event.preventDefault();
+
+    this.setState({
+      inProgress: true,
+      resetSuccess: false,
+      resetError: null
+    });
+
+    const token = this.props.location.query.reset_password_token;
+    const password = this.password.value;
+    const confirmation = this.confirmation.value;
+
+    auth.resetPassword({ password, confirmation, token })
+      .then(() => {
+        this.setState({
+          resetSuccess: true,
+          inProgress: false
+        });
+        alert(resolve => <LoginDialog onSuccess={resolve} />);
+        this.context.router.push('/projects');
+      })
+      .catch((error) => {
+        this.setState({ resetError: error });
+      });
+  }
+
+  renderNewPasswordForm() {
```

Comment:
Maybe just make this its own functional component?

Reply:
The partials folder should go away eventually. New components shouldn't go into that folder and existing components in there should move closer to where they're actually used. This folder is left over from the organizational structure when the app first launched, but it doesn't make sense anymore. 

The components folder is intended for reusable components.

I'd prefer them as separate files. You could make a folder for this page and its set of components for organizational purposes.